docs/core_modules/supporting_modules/service_context.md

```diff
@@ -13,12 +13,16 @@ The `ServiceContext` is a simple python dataclass that you can directly construc
 @dataclass
 class ServiceContext:
     # The LLM used to generate natural language responses to queries.
+    # If not provided, defaults to gpt-3.5-turbo from OpenAI
+    # If your OpenAI key is not set, defaults to llama2-chat-13B from Llama.cpp
     llm: LLM
 
     # The PromptHelper object that helps with truncating and repacking text chunks to fit in the LLM's context window.
     prompt_helper: PromptHelper
 
     # The embedding model used to generate vector representations of text.
+    # If not provided, defaults to text-embedding-ada-002
```

Comment:
curious: since `mpnet` model is trained on sentences, and meant to be used on sentences or short paragraphs, should we be setting smaller default chunk sizes (either just for local model or for all models)?

Reply:
Changing the chunk size under the user's nose feels a little wrong? But maybe if we print a warning it's fine?

In my testing though it seems to work fine enough for the current chunk size. If we lower the chunk size further, do we need to adjust the default top k too?

Reply:
according to [the model card](https://huggingface.co/sentence-transformers/all-mpnet-base-v2), sentence transformer truncates the input sentence up to some max length, in this case, looks like 384 tokens.

Doesn't this imply more than half of the chunk would be discarded?

Reply:
Actually it's 514 https://huggingface.co/sentence-transformers/all-mpnet-base-v2/blob/main/config.json

Similar reporting on the MTEB leaderboard https://huggingface.co/spaces/mteb/leaderboard
(LOL it ranks pretty low at #25. Would probably replace this with BGE once I write the integration for it)

Reply:
idk, I can change the chunk size I suppose if this model is automatically activated. OpenAI is the only one supporting chunk sizes larger than ~512 for embeddings 🙄

Reply:
![Screenshot 2023-08-10 at 5 13 26 PM](https://github.com/jerryjliu/llama_index/assets/5567282/22a8c0c6-64de-43ac-bb4a-99993f3701ae)

Reply:
dang, that sucks. How does it work so well LOL

Reply:
That's why we've been directing people to use every time they want local embeddings too...

Reply:
really need that BGE integration I guess lol

Reply:
chatted with Simon a bit on this. Ideal solution would be having an entire config yaml for different profiles for the service context (openai, local, anthropic, google, etc.)